File: C,C++/p100.c
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

#include <stdio.h>

int cekCycle (int);

int main () {
    int num1, num2;
    while(scanf("%d %d", &num1, &num2) != EOF) {
        int res = 0;
        int i;
        if (num1 < num2) {
        	for (i = num1; i <= num2; i++) {
            	int temp = cekCycle(i);
            	if (temp > res) res = temp;
        	}
		} else {
			for (i = num2; i <= num1; i++) {
            	int temp = cekCycle(i);
            	if (temp > res) res = temp;
        	}
		}
        
        
        printf("%d %d %d\n", num1, num2, res);
    }
    
    return 0;
}

int cekCycle (int num) {
    int res = 1;
    while (num != 1) {
        if (num % 2 == 1) num = 3*num + 1;
        else num/=2;
        res++;
    }
    
    return res;
}

File: Java/p100.java

import java.util.Scanner;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author DarKnight060198
 */
public class p100 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        while(input.hasNext()){
            int number1 = input.nextInt();
            int number2 = input.nextInt();

            int num1 = 0, num2 = 0;

            if (number1 > number2) { 
                num1 = number2;
                num2 = number1;
            } else {
                num1 = number1;
                num2 = number2;
            }

            int total = 0;

            for (int i = num1; i <= num2; i++) {
                int res = 1;
                int temp = i;
                while (temp != 1) {
                    if (temp%2 == 1) temp = temp*3 + 1;
                    else temp /= 2;
                    res++;
                }
                if (res > total) total = res;
            }

            System.out.printf("%d %d %d\n", number1, number2, total);
        }
    }
}
